src/org/opensolaris/opengrok/configuration/Configuration.java

```diff
@@ -786,7 +786,9 @@ public void setCurrentIndexedCollapseThreshold(int currentIndexedCollapseThresho
     public Date getDateForLastIndexRun() {
         if (lastModified == null) {
             File timestamp = new File(getDataRoot(), "timestamp");
-            lastModified = new Date(timestamp.lastModified());
+            if(timestamp.exists()){
```

Comment:
Without file existence check, date will be set as Jan 01 1970, while it should be null, this causes bogus behaviour